dlt/destinations/impl/duckdb/factory.py

```diff
@@ -0,0 +1,31 @@
+import typing as t
+
+from dlt.common.configuration import with_config, known_sections
+from dlt.common.destination.reference import DestinationClientConfiguration, Destination
+
+from dlt.destinations.impl.duckdb.configuration import DuckDbCredentials, DuckDbClientConfiguration
+from dlt.destinations.impl.duckdb import capabilities
+
+if t.TYPE_CHECKING:
+    from dlt.destinations.impl.duckdb.duck import DuckDbClient
+
+
+class duckdb(Destination):
+
+    capabilities = capabilities()
+    spec = DuckDbClientConfiguration
+
+    @property
+    def client_class(self) -> t.Type["DuckDbClient"]:
+        from dlt.destinations.impl.duckdb.duck import DuckDbClient
+
+        return DuckDbClient
+
+    @with_config(spec=DuckDbClientConfiguration, sections=(known_sections.DESTINATION, 'duckdb'), accept_partial=True)
```

Comment:
OK, we can simplify this even more. 
1. I'd not use @with_config here at. I'd just let people pass initial parameters and do not pull config at all.
2. I'd use with_config not as decorator but within a function that would return a configuration. there's no partial.
so let's imagine we have this in `Destination`:
```
def configuration(initial: TConfig) -> TConfig:
   return with_config(lambda x: x, spec=DuckDbClientConfiguration, sections=(known_sections.DESTINATION, self.destination_name), initial_config=initial)
```
we'd need to extend `with_config` to accept inital config in the decorator BUT if you look in the code there's already mechanims to handle that:
```
# if one of arguments is spec the use it as initial value
                if spec_arg:
                    config = bound_args.arguments.get(spec_arg.name, None)                    
```
if we do that, we do not use partial, we always resolve configuration etc.
where `initial` is coming from? we form it in `__init__` from the aguments.

Reply:
Hmm yeah I think that makes sense. So we'd just be leaving resolving config until the last moment when it's needed.

Reply:
Question, why use `with_config` and not `resolve_config` directly for this?

Reply:
you can use `resolve_config`! looks even better